src/Controller/Controller.php

```diff
@@ -694,29 +694,6 @@ public function redirect($url, int $status = 302): ?Response
         return $this->response = $response;
     }
 
-    /**
-     * Internally redirects one action to another. Does not perform another HTTP request unlike Controller::redirect()
-     *
-     * Examples:
-     *
-     * ```
-     * setAction('another_action');
-     * setAction('action_with_parameters', $parameter1);
-     * ```
-     *
-     * @param string $action The new action to be 'redirected' to.
-     *   Any other parameters passed to this method will be passed as parameters to the new action.
-     * @param mixed ...$args Arguments passed to the action
-     * @return mixed Returns the return value of the called action
-     * @deprecated 4.2.0 Refactor your code use `redirect()` instead of forwarding actions.
-     */
-    public function setAction(string $action, ...$args)
-    {
-        $this->setRequest($this->request->withParam('action', $action));
-
-        return $this->$action(...$args);
-    }
```

Comment:
better to have `deprecationWarning()` in cake 4.x before removing this method, IMO